RecoTauTag/HLTProducers/src/L2TauTagNNProducer.cc

```diff
@@ -567,31 +566,33 @@ void L2TauNNProducer::fillCaloRecHits(tensorflow::Tensor& cellGridMatrix,
 }
 
 std::vector<int> L2TauNNProducer::selectGoodVertices(const ZVertexSoA& patavtx_soa,
-                                                     const pixelTrack::TrackSoA& patatracks_tsoa,
-                                                     const std::vector<int>& TrackGood) {
-  auto maxTracks = patatracks_tsoa.stride();
-  const int nv = patavtx_soa.nvFinal;
-  std::vector<int> VtxGood;
-  if (nv == 0)
-    return VtxGood;
-  VtxGood.reserve(nv);
-
-  std::vector<double> maxChi2_;
-  std::vector<double> pTSquaredSum(nv);
-
-  for (int j = nv - 1; j >= 0; --j) {
-    std::vector<int> trk_ass_to_vtx;
-    auto vtx_idx = patavtx_soa.sortInd[j];
-    assert(vtx_idx < nv);
-    for (int trk_idx = 0; trk_idx < maxTracks; trk_idx++) {
-      int vtx_ass_to_track = patavtx_soa.idv[trk_idx];
-      if (vtx_ass_to_track == int16_t(vtx_idx))
-        trk_ass_to_vtx.push_back(trk_idx);
-    }
-    auto nt = trk_ass_to_vtx.size();
-    if (nt == 0) {
-      continue;
-    }
+                                                     const pixelTrack::TrackSoA& patatracks_tsoa) {
+   auto maxTracks = patatracks_tsoa.stride();
+   const int nv = patavtx_soa.nvFinal;
+   std::vector<int> VtxGood;
+   if (nv == 0)
+     return VtxGood;
+   VtxGood.reserve(nv);
+
+   std::vector<double> maxChi2_;
+   std::vector<double> pTSquaredSum(nv);
+
+   for (int j = nv - 1; j >= 0; --j) {
```

Comment:
this double loop is not required.
enough to loop on the tracks, apply selection, and fill pTSquaredSum directly.
No need to sort either as the algorithms is just using the max (not even the location, just the max value of pt2sum).